File: src/main/java/com/nugurang/graphql/query/ImageQuery.java
package com.nugurang.graphql.query;

import com.nugurang.dto.ImageDto;
import com.nugurang.exception.NotFoundException;
import com.nugurang.service.ImageService;
import graphql.kickstart.tools.GraphQLQueryResolver;
import java.util.Optional;
import org.springframework.stereotype.Service;

@Service
public class ImageQuery implements GraphQLQueryResolver {
    private final ImageService imageService;

    public Optional<ImageDto> getImage(Long id) throws NotFoundException {
        return Optional.ofNullable(imageService.getImage(id)).map(entity -> entity.toDto());
    }

    public Optional<ImageDto> getImageByAddress(String address) throws NotFoundException {
        return Optional.ofNullable(imageService.getImage(address)).map(entity -> entity.toDto());
    }

    //<editor-fold defaultstate="collapsed" desc="delombok">
    @SuppressWarnings("all")
    
    public ImageQuery(final ImageService imageService) {
        this.imageService = imageService;
    }
    //</editor-fold>
}

File: src/main/kotlin/com/nugurang/graphql/query/ImageQuery.kt
package com.nugurang.graphql.query

import com.nugurang.dto.ImageDto
import com.nugurang.mapper.ImageMapper
import com.nugurang.service.ImageService
import graphql.kickstart.tools.GraphQLQueryResolver
import org.springframework.stereotype.Service

@Service
class ImageQuery(
    private val imageService: ImageService,
    private val imageMapper: ImageMapper
) : GraphQLQueryResolver {

    fun getImage(id: Long): ImageDto {
        return imageService.getImage(id).let(imageMapper::toDto)
    }

    fun getImageByAddress(address: String): ImageDto {
        return imageService.getImage(address).let(imageMapper::toDto)
    }
}